cdap-unit-test/src/main/java/co/cask/cdap/test/internal/DefaultApplicationManager.java

```diff
@@ -118,14 +116,18 @@ public WorkerManager getWorkerManager(String workerName) {
   @Override
   public void stopAll() {
     try {
-      for (ProgramId programId : Iterables.consumingIterable(runningProcesses)) {
-        // have to do a check, since mapreduce jobs could stop by themselves earlier, and appFabricServer.stop will
-        // throw error when you stop something that is not running.
-        if (isRunning(programId)) {
-          appFabricClient.stopProgram(application.getNamespace(), programId.getApplication(),
-                                      programId.getProgram(), programId.getType());
+      ApplicationDetail appDetail = appFabricClient.getVersionedInfo(application);
+      for (ProgramRecord programRecord : appDetail.getPrograms()) {
+        try {
+          appFabricClient.stopProgram(application.getNamespace(), application.getApplication(),
+                                      appDetail.getAppVersion(), programRecord.getName(), programRecord.getType());
+        } catch (BadRequestException e) {
+          // Ignore this as this will be throw if the program is not running, which is fine as there could
+          // be program runs that was stopped by itself and still exists in the runningProcesses
```

Comment:
this comment about `runningProcesses` seems inaccurate now.
It seems like we call `stopProgram` for every program in the app, even if it wasn't started or if its not even running